src/constraints/battery_degradation.jl

```diff
@@ -42,6 +42,95 @@ function constrain_degradation_variables(m, p; b="ElectricStorage")
     )
 end
 
+## Charging and discharging power for each ts for each segment
+function add_segmented_cycle_fade_variables(m, p; b="ElectricStorage")
+
+    days = 1:365*p.s.financial.analysis_years
+    J = length(p.s.storage.attr[b].degradation.segment_energy_capacity); # Number of segments
+
+    @variable(m, p_c[p.time_steps, 1:J] >= 0) # charge power for each ts for each segment 
+    @variable(m, p_d[p.time_steps, 1:J] >= 0); # discharge power for each ts for each segment
+
+    @variable(m, e_plus[days, 1:J] >= 0); # energy added for each day for each segment level
+    @variable(m, e_minus[days, 1:J] >= 0); # energy discharged for each day for each segment level
+end
+
+## Charging and discharging power for each ts for each segment
+function constrain_segmented_cycle_fade_variables(m, p; b="ElectricStorage")
+    days = 1:365*p.s.financial.analysis_years
+    ts_per_day = 24 / p.hours_per_time_step
+    ts_per_year = ts_per_day * 365
+    ts0 = Dict()
+    tsF = Dict()
+    for d in days
+        ts0[d] = Int((ts_per_day * (d - 1) + 1) % ts_per_year)
+        tsF[d] = Int(ts_per_day * d % ts_per_year)
+        if tsF[d] == 0
+            tsF[d] = Int(ts_per_day * 365)
+        end
+    end
+
+    days = 1:365*p.s.financial.analysis_years
+    J = length(p.s.storage.attr[b].degradation.segment_energy_capacity); # Number of segments
+
+    # Power in equals power into storage from grid or local production. (question)
+    @constraint(m, [ts in p.time_steps],
+    sum(m[:p_c][ts, j] for j in 1:J) == sum(m[:dvProductionToStorage][b, t, ts] for t in p.techs.elec) 
+            + sum(m[:dvGridToStorage][b, ts]))   
+            #[az] last term in RHS is not a sum, it's a single element since b is defined and the constraint is indexed on ts.
+
+    # Power out equals power discharged from storage to any destination
+    @constraint(m, [ts in p.time_steps], sum(m[:p_d][ts, j] for j in 1:J) == sum(m[:dvDischargeFromStorage][b, ts]));
+    #[az] RHS is not a sum, it's a single element since b is defined and the constraint is indexed on ts.
+
+    # Balance charging with daily e_plus, here is only collect all power across the day, so don't need to times efficiency
+    @constraint(m, [d in days, j in 1:J], m[:e_plus][d, j] == sum(m[:p_c][ts0[d]:tsF[d], j])*p.hours_per_time_step)
+    @constraint(m, [d in days, j in 1:J], m[:e_minus][d, j] == sum(m[:p_d][ts0[d]:tsF[d], j])*p.hours_per_time_step);
+    #[az] we may want to adjust the notation to "ts, j for ts in ts0[d]:tsF[d] so it reads the same as the other constraints in REopt
+
+    # energy limit, replace SOC limitation
+    @constraint(
+        m,
+        [ts in p.time_steps, j in 1:J],
+        m[:p_c][ts, j]*p.hours_per_time_step <= p.s.storage.attr[b].degradation.segment_energy_capacity[j]*m[:dvStorageEnergy][b]
```

Comment:
is p_c a fraction?  We may want to clarify that as we describe the inputs and the bounds on these.  It's also confusing that we have a power variable bounded above by the energy capacity.

Reply:
ren segment_energy_capacity segment_energy_capacity_fraction?